api/v1alpha1/enterprisecontractpolicy_types.go

```diff
@@ -43,6 +43,9 @@ type EnterpriseContractPolicySpec struct {
 	// Public key used to validate the signature of images and attestations
 	// +optional
 	PublicKey string `json:"publicKey,omitempty"`
+	// A json or yaml string containing rule data that will be visible to policy rules
+	// +optional
+	RuleData string `json:"ruleData,omitempty"`
```

Comment:
Could use [`k8s.io/apiextensions-apiserver/pkg/apis/apiextensions.JSON`](https://pkg.go.dev/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions#JSON) here 
```suggestion
	RuleData ext.JSON `json:"ruleData,omitempty"`
```

Reply:
I wondered if there was such a thing. Would there be a YAML equivalent, or should we just say it needs to be json?

Reply:
Kubernetes API is JSON (or protobuf), the YAML aspect of it is only in the clients. This would allow any inline JSON, i.e. not a string containing JSON, and then by extension any YAML. I'm 90% sure on this, though I haven't found good source of documentation on this. We could test this via unit test of marshaling logic...

Reply:
Created #94 as with this approach